### PR TITLE
:sparkles: `withRootFiles` now apply rector for dotfiles too

### LINKED ISSUE #8896: withRootFiles() ignores ".php-cs-fixer.dist.php"
# Feature Request

Documentation at [1] states, that using `withRootFiles` will make sure, that rector will apply rules to the root directory PHP files like a "rector.php", "ecs.php", etc.

But rector doesn't apply rules for ".php-cs-fixer.dist.php" at the root directory for now.

Feature request is: `withRootFiles` should include hidden PHP files from the root (or at least for well known files like a ".php-cs-fixer.dist.php")

[1] https://getrector.com/documentation/define-paths

## Diff 

N/A

### PR DESCRIPTION
**Pull request by andrew-demb**
Resolves https://github.com/rectorphp/rector/issues/8896

### CODE REVIEW

**Comment by andrew-demb** [comment]
Side question: why is there a custom implementation of parsing the `.gitignore` file in `withRootFiles()`, but not using the `symfony/finder` feature?

https://symfony.com/doc/current/components/finder.html#:~:text=%24finder%2D%3E-,ignoreVCSIgnored,-(true)%3B 

> If the search directory and its subdirectories contain .gitignore files, you can reuse those rules to exclude files and directories from the results with the [ignoreVCSIgnored()](https://github.com/symfony/symfony/blob/7.2/src/Symfony/Component/Finder/Finder.php#:~:text=function%20ignoreVCSIgnored) method:
> ```
> // excludes files/directories matching the .gitignore patterns
> $finder->ignoreVCSIgnored(true);
> ```

File: .phpstorm.meta.php [hunk 1]
```diff
@@ -5,13 +5,17 @@
...
 
+use Psr\Container\ContainerInterface;
+use PHPStan\PhpDocParser\Ast\Node;
+use PHPStan\PhpDocParser\Ast\NodeAttributes;
+use Rector\NodeTypeResolver\Node\AttributeKey;
 use Rector\BetterPhpDocParser\ValueObject\PhpDocAttributeKey;
 
 // $container->get(Type::class) → instance of "Type"
-override(\Psr\Container\ContainerInterface::get(0), type(0));
+override((new ContainerInterface())->get(0), type(0));
```
Comment (TomasVotruba) [inline .phpstorm.meta.php]: Could you split these meta changes to its own PR? Just so we have changes independent in case of blame/regression

Comment (andrew-demb) [reply .phpstorm.meta.php]: I think, that without changing this file, there would be a failed CI for the current PR.

Here's the separate PR https://github.com/rectorphp/rector-src/pull/6593

**Comment by TomasVotruba** [comment]
Thank you, let's give this a try :+1:
